Add tailscale docs (#142)
k3s-io/docs: docs/installation/network-options.md

--cluster-cidr=2001:cafe:42:0::/56 --service-cidr=2001:cafe:42:1::/112
```

## Distributed hybrid or multicloud cluster

> A K3s cluster can still be deployed on nodes which use different private networks and are not directly connected (e.g. nodes in different public clouds). There are two options to achieve this: wireguard-native and tailscale

Reviewer: ```suggestion
A K3s cluster can be deployed on nodes which do not share a common private network and are not directly connected (e.g. nodes in different public clouds). There are two options to achieve this: the `wireguard-native` flannel backend, and the `tailscale` VPN provider.
```

Author: This is not accurate though, because one option is not `wireguard-native`. That is what pod traffic is using, however, this is not what control-planes are using. They use public IP + https to communicate. So, summarizing the solution by saying `wireguard-native` is misleading. I think we should find a better naming, something like: 
```
There are two options to achieve this: the k3s internal multicloud solution, and the integration with tailscale VPN provider.
```
And then describe both

:::caution Warning
The latency between nodes will increase as external connectivity requires more hops. This will reduce the network performance and could also impact the health of the cluster if latency is too high.
:::

:::caution Warning

> Embedded etcd is not supported in this type of deployment. If using embedded etcd; all server nodes must be reachable to each other via their private IPs.

Reviewer: ```suggestion
If using embedded etcd; all server nodes must be reachable to each other via their private IPs. Agents may be distributed over multiple networks, but all servers should be in the same location.
```

Author: ok, thanks!

:::

### Wireguard-native

> K3s can set a mesh of tunnels that become a VPN mesh. Nodes must have have an assigned IP through which they can be reached (e.g. a public IP). The server traffic will use a websocket tunnel and the data-plane traffic will use a wireguard tunnel.

Reviewer: ```suggestion
K3s use wireguard to establish a VPN mesh for cluster traffic. Nodes must each have a unique IP through which they can be reached (usually a public IP). K3s supervisor traffic will use a websocket tunnel, and cluster (CNI) traffic will use a wireguard tunnel.
```

Author: ok

```
:::

### Tailscale

Available in v1.27.3, v1.26.6, v1.25.11 and newer.

> K3s can integrate with [tailscale](https://tailscale.com/) so that nodes use the tailscale VPN network to communicate with each other.

Reviewer: ```suggestion
K3s can integrate with [Tailscale](https://tailscale.com/) so that nodes use the Tailscale VPN service to build a mesh between nodes.
```

Author: good, thanks!

> There are four steps to be done with tailscale before deploying k3s:
> 1 - Create an account in tailscale

Reviewer: ```suggestion
There are four steps to be done with Tailscale before deploying k3s:
1 - Log in to your Tailscale account
```

Author: English question: `Log in to` or `log into`

Reviewer: Maybe "Sign in"? I have seen it as "login to", " log in to", and "log into", but I think only the three word version is grammatically correct.

> 2 - In Settings > Keys, generate an auth key (<auth-key>), which may be reusable for all nodes in your cluster

Reviewer: Does this need to be unique per cluster? Can multiple clusters use the same key? Are there any other considerations for having multiple clusters on a single account?

3 - Decide on the podCIDR the cluster will use (by default 10.42.0.0/16). Append the CIDR (or CIDRs for dual-stack) in Access controls with the stanza:
```yaml
"autoApprovers": {
		"routes": {
			"10.42.0.0/16":        ["your_account@xyz.com"],
			"2001:cafe:42:0::/56": ["your_account@xyz.com"],
		},
	},
```

> 4 - Install tailscale in your nodes:

Reviewer: ```suggestion
4 - Install Tailscale in your nodes:
```

```bash
curl -fsSL https://tailscale.com/install.sh | sh
```

> To deploy k3s with tailscale integration enabled, you must add the following parameter on each of your nodes:

Reviewer: ```suggestion
To deploy k3s with Tailscale integration enabled, you must add the following parameter on each of your nodes:
```

Author: :+1:
